server/app/views/admin/programs/ProgramBlockEditView.java

```diff
@@ -878,6 +881,82 @@ private DivTag questionBankPanel(
     return qb.getContainer(questionBankVisibility);
   }
 
+  private Modal blockDeleteModal(
+      InputTag csrfTag, String blockDeleteAction, BlockDefinition blockDefinition) {
+
+    FormTag deleteBlockForm =
+        form(csrfTag)
+            .withId(DELETE_BLOCK_FORM_ID)
+            .withMethod(HttpVerbs.POST)
+            .withAction(blockDeleteAction);
+
+    boolean hasQuestions = blockDefinition.getQuestionCount() > 0;
+    boolean hasEligibilityCondition = !blockDefinition.eligibilityDefinition().isEmpty();
+    boolean hasVisibilityCondition = !blockDefinition.visibilityPredicate().isEmpty();
+    ArrayList<String> itemsInBlock = new ArrayList<String>();
+
+    if (hasQuestions) {
+      itemsInBlock.add("questions");
+    }
+    if (hasEligibilityCondition) {
+      itemsInBlock.add("eligibility conditions");
+    }
+    if (hasVisibilityCondition) {
+      itemsInBlock.add("visibility conditions");
+    }
+
+    // If there are no questions, eligibilty conditions, or visibility conditions on this screen,
+    // just print "Are you sure you want to delete this screen?"
+    if (itemsInBlock.size() == 0) {
+      deleteBlockForm
+          .withId("block-delete-form")
+          .with(
+              div(h1("Are you sure you want to delete this screen?")
+                      .withClasses("text-base", "mb-2"))
+                  .withClasses("mx-4"),
+              submitButton("Delete")
+                  .withId("delete-block-button")
+                  .withClasses(
+                      "mx-4", "my-1", "inline", "opacity-100", StyleUtils.disabled("opacity-50")));
+    } else {
+      // If there are questions, eligibility conditions, or visibility conditions on this screen,
+      // print the appropriate message.
+      String listItemsInBlock = "";
+      if (itemsInBlock.size() == 1) {
+        listItemsInBlock = itemsInBlock.get(0);
+      } else if (itemsInBlock.size() == 2) {
+        listItemsInBlock = itemsInBlock.get(0) + " and " + itemsInBlock.get(1);
+      } else {
+        // there will only ever be 1, 2, or 3 items
+        listItemsInBlock =
+            itemsInBlock.get(0) + ", " + itemsInBlock.get(1) + " and " + itemsInBlock.get(2);
```

Comment:
**thought:** this makes me miss rails 😆 
https://apidock.com/rails/ActiveSupport/CoreExtensions/Array/Conversions/to_sentence

Reply:
😱 wow this would be soooo nice